flexget/components/notify/notifiers/gotify.py

```diff
@@ -0,0 +1,104 @@
+import logging
+import base64
+import datetime
+import re
+
+from flexget import plugin
+from flexget.event import event
+from flexget.config_schema import one_or_more
+from flexget.plugin import PluginWarning
+from flexget.utils.requests import Session as RequestSession, TimedLimiter
+from requests.exceptions import RequestException
+
+plugin_name = 'gotify'
+log = logging.getLogger(plugin_name)
+
+requests = RequestSession(max_retries=3)
+
+gotify_url_pattern = {
+    'type': 'string',
+    'pattern': r'^http(s?)\:\/\/.*\/message$',
+    'error_pattern': 'Gotify URL must begin with http(s) and end with `/message`',
+}
+
+class GotifyNotifier(object):
+    """
+    Example::
+
+    notify:
+      entries:
+        via:
+          - gotify:
+              url: <GOTIFY_SERVER_URL>
+              token: <GOTIFY_TOKEN>
+              priority: <PRIORITY>
+    Configuration parameters are also supported from entries (eg. through set).
+    """
+    schema = {
+        'type': 'object',
+        'properties': {
+            'url': one_or_more(gotify_url_pattern),
+            'token': {'type': 'string'},
+            'priority': {'type': 'integer', 'default': 4},  
+        },  
+        'required': [
+            'token',
+            'url',
+        ],
+        'anyOf': [
```

Comment:
i'm a little confused about the schema here. Looks like you enforced both `token` and `url` as required but then only enforce that one of them is? along with priority?

Reply:
I struggled with this a bit. Basically `token` and `url` are required, and `priority` is optional, but only one of each should be declared under `gotify` in the config. I'm guessing there's a better way to structure this?

Reply:
Just set url and token as required. You cant pass more than one keyword anyway since this is an object (dict)

Reply:
Should be resolved with https://github.com/4oo4/Flexget/commit/7e1b2e359dff01e68444f179852b3dd4d393bb3c